### DISCUSSION TITLE
compile.py, "error: static assertion failed"

### DISCUSSION THREAD

**Discussion by Xie-ry** [intel/llvm#12572] [Q&A]
Following GetStartedGuide.md, i use compile.py to build DPC++ toolchain.
Towards completion, I got the error:

> /data/user/root/llvm-sycl/sycl/source/builtins/host_helper_macros.hpp:42:24: error: static assertion failed
   42 |     static_assert(std::is_same_v<decltype(NAME##_host_impl(NUM_ARGS##_ARG)),   \
      |                   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   43 |                                  decltype(NS::NAME(NUM_ARGS##_ARG))>);         \

I tried a Google search and couldn't find a solution. Can someone help me?

I use gcc 9.1.0, ninja v1.11, Cmake 3.25

**Comment by Xie-ry** [comment]
--no-assertions is not work

**Reply by bader** [reply]
@Xie-ry, could you remove "build" directory, run configure.py and compile.py and attach the full output from these scripts?

**Reply by Xie-ry** [reply]
I try a old version solved my propblem
